### ISSUE TITLE
False negative on "unused return values" ?

### ISSUE THREAD

**Issue by kbaird** [rrrene/credo#36]
Hi. The code at https://github.com/mattmills/ecto_neo4j/pull/4 reports a warning:

```elixir
┃ [W] ↗ There should be no unused return values for Enum.map().
┃       lib/ecto_neo4j.ex:61:15 (Ecto.Neo4j.sort_column)
```

That I believe is incorrect. The context is a value returned from an if check being the last evaluated expression in a function:

```elixir
    if hd(cols) do
      coercer = fn({name, type}) -> coerce(type, col[Atom.to_string(name)]) end 
      cols |> Enum.map(coercer)
    else
      [nil]
    end
```

It looks like the presence of the `[nil]` and `end` lines below the `Enum.map/2` is confusing it into thinking that there are additional expressions leading the `map` value to be discarded.

**Comment by rrrene** [comment]
Thanks for reporting this, it indeed looks like a bug! :+1:

I will look into it!

**Comment by rrrene** [comment]
Bugfix is live in `v0.2.6` - Thanks again Kevin for a comprehensive bug report. Open Source at its best! :+1:

**Comment by kbaird** [comment]
Thanks for such a quick fix.
